### PR TITLE
Remove types from comments.

### PR DESCRIPTION
**Pull request by clokep**
These always annoy me. I came up with a regex to find them:

```bash
$ grep -rE "^ +[a-zA-Z_]+ \(.+\) *[:$]" synapse tests | grep -vE ": +(if|for|except) "
$ grep -rE -A1 "^ +Returns?:?$" synapse tests | grep -E "\- +[a-zA-Z].+:"
```

This doesn't clean-up all instance, namely it doesn't touch ones that refer to class args. Those seemed somewhat useful still? At least in some cases.

It also adds a few missing types where it seemed easier to just fill them in from the comments.

### CODE REVIEW

**Comment by clokep** [comment]
Oh hmm old deps is failing here. Will need to investigate.

File: tests/replication/slave/storage/test_events.py
Comment (clokep) [reply tests/replication/slave/storage/test_events.py:285]: Without this, we get a mypy error. I think I might have put the assertion too early though. 👍

File: synapse/handlers/e2e_room_keys.py
Comment (DMRobertson) [inline synapse/handlers/e2e_room_keys.py:381]: What does `version=None` mean here---looks like it means "the most recent backup"?

(FWIW my guess would have been "delete all versions")

Comment (clokep) [reply synapse/handlers/e2e_room_keys.py:381]: This copies the docstring from `delete_e2e_room_keys_version`:

```suggestion
            version: Optional. the version ID of the backup version we're deleting
                If missing, we delete the current backup version info.
```

File: tests/unittest.py
Comment (clokep) [reply tests/unittest.py:363]: This makes a lot of things fail (I had tried it before putting up this PR). IMO it is not worth holding up this PR to add that.

File: synapse/server_notices/consent_server_notices.py
Comment (clokep) [reply synapse/server_notices/consent_server_notices.py:117]: Maybe, but my goal in this PR wasn't to get rid of `Any`.

File: synapse/storage/databases/main/e2e_room_keys.py
Comment (DMRobertson) [reply synapse/storage/databases/main/e2e_room_keys.py:397]: Oh I see. Thanks.

File: synapse/storage/databases/main/room.py
Comment (clokep) [reply synapse/storage/databases/main/room.py:2060]: I think so? Not sure why this said `json list`...

```suggestion
            JSON dict of information from an event report or None if it
            does not exist.
```

File: synapse/handlers/saml.py
Comment (clokep) [reply synapse/handlers/saml.py:447]: Agreed, but I'm going to leave that to the future.

**Review by DMRobertson — APPROVED** [review]

**Comment by clokep** [comment]
Thanks for the review! 🎉

**Pull request merged**
